### PR TITLE
Add test for scaffold.0.dart and scaffold.2.dart

### PR DESCRIPTION
**Pull request by ValentinVignal**
Contributes to https://github.com/flutter/flutter/issues/130459

It adds test for
- `examples/api/lib/material/scaffold/scaffold.0.dart`
- `examples/api/lib/material/scaffold/scaffold.2.dart`

It also modifies the `scaffold.1_test.dart` because the only difference with scaffold 0 is the background color, so I figured the test should include it

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by ValentinVignal** [comment]
I don't understand this comment about the changing goldens. the tests I  modified / created don't contain any golden match, and I didn't modify any flutter code. 
What am I supposed to do/not do ?

File: examples/api/test/material/scaffold/scaffold.2_test.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    expect(find.text('You have pressed the button 0 times.'), findsOne);
+    expect(find.byType(BottomAppBar), findsOne);
+    expect(tester.getCenter(find.byType(FloatingActionButton)).dx, 400);
+
+    for (int i = 1; i <= 5; i++) {
+      await tester.tap(find.byType(FloatingActionButton));
+      await tester.pump();
+      expect(find.text('You have pressed the button $i times.'), findsOne);
+    }
+
```
Comment (bleroux) [inline examples/api/test/material/scaffold/scaffold.2_test.dart:26]: Consider removing this empty line.

**Review by bleroux — APPROVED** [review]
LGTM! Thanks for the contribution.
I just left one comment related to an extra line.

File: examples/api/test/material/scaffold/scaffold.2_test.dart (hunk 1 above)
Comment (TahaTesser) [inline examples/api/test/material/scaffold/scaffold.2_test.dart:26]:
```suggestion
```

**Review by TahaTesser — APPROVED** [review]
LGTM with nit

**Pull request merged**